In [3]:
import pandas as pd

In [6]:
data = pd.read_csv("CreditCardAnalytics.csv", sep="|")
data

CUST_ID  Bilanci_aktual  Frekuenca_ndryshimit_bilancit  \
0     C10001       40.900749                       0.818182   
1     C10002     3202.467416                       0.909091   
2     C10003     2495.148862                       1.000000   
3     C10004     1666.670542                       0.636364   
4     C10005      817.714335                       1.000000   
...      ...             ...                            ...   
8945  C19186       28.493517                       1.000000   
8946  C19187       19.183215                       1.000000   
8947  C19188       23.398673                       0.833333   
8948  C19189       13.457564                       0.833333   
8949  C19190      372.708075                       0.666667   

      Shuma_blerjeve_nga_llogaria  Shuma_transaksionit_metemadh  \
0                           95.40                          0.00   
1                            0.00                          0.00   
2                          773.17                        773.17   
3                         1499.00                       1499.00   
4                           16.00                         16.00   
...                           ...                           ...   
8945                       291.12                          0.00   
8946                       300.00                          0.00   
8947                       144.40                          0.00   
8948                         0.00                          0.00   
8949                      1093.25                       1093.25   

      Shuma_si_kest  Frekuenca_perdorimit_cc  Numri_blerjeve_cc  CREDIT_LIMIT  \
0             95.40                 0.166667                  2        1000.0   
1              0.00                 0.000000                  0        7000.0   
2              0.00                 1.000000                 12        7500.0   
3              0.00                 0.083333                  1        7500.0   
4              0.00                 0.083333                  1        1200.0   
...             ...                      ...                ...           ...   
8945         291.12                 1.000000                  6        1000.0   
8946         300.00                 1.000000                  6        1000.0   
8947         144.40                 0.833333                  5        1000.0   
8948           0.00                 0.000000                  0         500.0   
8949           0.00                 0.666667                 23        1200.0   

      Pjesa_paguar_limitit  Pagesa_minimale_cdomuaj_cc  Kohezgjatja_cc  
0               201.802084                  139.509787              12  
1              4103.032597                 1072.340217              12  
2               622.066742                  627.284787              12  
3                 0.000000                         NaN              12  
4               678.334763                  244.791237              12  
...                    ...                         ...             ...  
8945            325.594462                   48.886365               6  
8946            275.861322                         NaN               6  
8947             81.270775                   82.418369               6  
8948             52.549959                   55.755628               6  
8949             63.165404                   88.288956               6  

[8950 rows x 12 columns]

In [9]:
data.describe()

Bilanci_aktual  Frekuenca_ndryshimit_bilancit  \
count     8950.000000                    8950.000000   
mean      1564.474828                       0.877271   
std       2081.531879                       0.236904   
min          0.000000                       0.000000   
25%        128.281915                       0.888889   
50%        873.385231                       1.000000   
75%       2054.140036                       1.000000   
max      19043.138560                       1.000000   

       Shuma_blerjeve_nga_llogaria  Shuma_transaksionit_metemadh  \
count                  8950.000000                   8950.000000   
mean                   1003.204834                    592.437371   
std                    2136.634782                   1659.887917   
min                       0.000000                      0.000000   
25%                      39.635000                      0.000000   
50%                     361.280000                     38.000000   
75%                    1110.130000                    577.405000   
max                   49039.570000                  40761.250000   

       Shuma_si_kest  Frekuenca_perdorimit_cc  Numri_blerjeve_cc  \
count    8950.000000              8950.000000        8950.000000   
mean      411.067645                 0.490351          14.709832   
std       904.338115                 0.401371          24.857649   
min         0.000000                 0.000000           0.000000   
25%         0.000000                 0.083333           1.000000   
50%        89.000000                 0.500000           7.000000   
75%       468.637500                 0.916667          17.000000   
max     22500.000000                 1.000000         358.000000   

       CREDIT_LIMIT  Pjesa_paguar_limitit  Pagesa_minimale_cdomuaj_cc  \
count   8949.000000           8950.000000                 8637.000000   
mean    4494.449450           1733.143852                  864.206542   
std     3638.815725           2895.063757                 2372.446607   
min       50.000000              0.000000                    0.019163   
25%     1600.000000            383.276166                  169.123707   
50%     3000.000000            856.901546                  312.343947   
75%     6500.000000           1901.134317                  825.485459   
max    30000.000000          50721.483360                76406.207520   

       Kohezgjatja_cc  
count     8950.000000  
mean        11.517318  
std          1.338331  
min          6.000000  
25%         12.000000  
50%         12.000000  
75%         12.000000  
max         12.000000

In [8]:
data.isnull().sum()

CUST_ID                            0
Bilanci_aktual                     0
Frekuenca_ndryshimit_bilancit      0
Shuma_blerjeve_nga_llogaria        0
Shuma_transaksionit_metemadh       0
Shuma_si_kest                      0
Frekuenca_perdorimit_cc            0
Numri_blerjeve_cc                  0
CREDIT_LIMIT                       1
Pjesa_paguar_limitit               0
Pagesa_minimale_cdomuaj_cc       313
Kohezgjatja_cc                     0
dtype: int64

We see that we have some null values in the colun 'Pagesa_minimale_cdo_muaj_cc.
Also the column Credit_Limit has null values


In [10]:
data.dropna()

CUST_ID  Bilanci_aktual  Frekuenca_ndryshimit_bilancit  \
0     C10001       40.900749                       0.818182   
1     C10002     3202.467416                       0.909091   
2     C10003     2495.148862                       1.000000   
4     C10005      817.714335                       1.000000   
5     C10006     1809.828751                       1.000000   
...      ...             ...                            ...   
8943  C19184        5.871712                       0.500000   
8945  C19186       28.493517                       1.000000   
8947  C19188       23.398673                       0.833333   
8948  C19189       13.457564                       0.833333   
8949  C19190      372.708075                       0.666667   

      Shuma_blerjeve_nga_llogaria  Shuma_transaksionit_metemadh  \
0                           95.40                          0.00   
1                            0.00                          0.00   
2                          773.17                        773.17   
4                           16.00                         16.00   
5                         1333.28                          0.00   
...                           ...                           ...   
8943                        20.90                         20.90   
8945                       291.12                          0.00   
8947                       144.40                          0.00   
8948                         0.00                          0.00   
8949                      1093.25                       1093.25   

      Shuma_si_kest  Frekuenca_perdorimit_cc  Numri_blerjeve_cc  CREDIT_LIMIT  \
0             95.40                 0.166667                  2        1000.0   
1              0.00                 0.000000                  0        7000.0   
2              0.00                 1.000000                 12        7500.0   
4              0.00                 0.083333                  1        1200.0   
5           1333.28                 0.666667                  8        1800.0   
...             ...                      ...                ...           ...   
8943           0.00                 0.166667                  1         500.0   
8945         291.12                 1.000000                  6        1000.0   
8947         144.40                 0.833333                  5        1000.0   
8948           0.00                 0.000000                  0         500.0   
8949           0.00                 0.666667                 23        1200.0   

      Pjesa_paguar_limitit  Pagesa_minimale_cdomuaj_cc  Kohezgjatja_cc  
0               201.802084                  139.509787              12  
1              4103.032597                 1072.340217              12  
2               622.066742                  627.284787              12  
4               678.334763                  244.791237              12  
5              1400.057770                 2407.246035              12  
...                    ...                         ...             ...  
8943             58.644883                   43.473717               6  
8945            325.594462                   48.886365               6  
8947             81.270775                   82.418369               6  
8948             52.549959                   55.755628               6  
8949             63.165404                   88.288956               6  

[8636 rows x 12 columns]

In [41]:
data.nunique()

CUST_ID                          8950
Bilanci_aktual                   8871
Frekuenca_ndryshimit_bilancit      43
Shuma_blerjeve_nga_llogaria      6203
Shuma_transaksionit_metemadh     4014
Shuma_si_kest                    4452
Frekuenca_perdorimit_cc            47
Numri_blerjeve_cc                 173
CREDIT_LIMIT                      205
Pjesa_paguar_limitit             8711
Pagesa_minimale_cdomuaj_cc       8636
Kohezgjatja_cc                      7
dtype: int64

In [42]:
data['Shuma_transaksionit_metemadh'].sort_index().head(10)

0       0.00
1       0.00
2     773.17
3    1499.00
4      16.00
5       0.00
6    6402.63
7       0.00
8     661.49
9    1281.60
Name: Shuma_transaksionit_metemadh, dtype: float64

This shows top 10 most transaction number

In [43]:
data['Shuma_transaksionit_metemadh'].count()

8950

Total number of transactions

In [44]:
data.groupby('Kohezgjatja_cc')['Numri_blerjeve_cc'].sum()

Kohezgjatja_cc
6       1161
7       1077
8       1225
9       1150
10      1847
11      2868
12    122325
Name: Numri_blerjeve_cc, dtype: int64

So we have based on length of the agrement of each item buyed, for example for length of 6 month number of things bought are 1161, while length of 12 months number of items bought are 122325. This shows that people prefer to buy stuff with loan in more long term of months > 6 months to returning the money.

In [45]:
data.groupby('CUST_ID')['Numri_blerjeve_cc'].sum()

CUST_ID
C10001     2
C10002     0
C10003    12
C10004     1
C10005     1
          ..
C19186     6
C19187     6
C19188     5
C19189     0
C19190    23
Name: Numri_blerjeve_cc, Length: 8950, dtype: int64

While this shows the items bought based on costumer id. Now lets see top 10 costumers that spend more money or use CC the most.

In [46]:
top10 = data.groupby('CUST_ID')['Numri_blerjeve_cc'].sum().sort_values(ascending=False).head(10)
top10

CUST_ID
C11300    358
C13272    347
C14048    344
C14918    309
C14640    308
C12905    298
C15731    274
C11974    273
C12589    254
C15190    248
Name: Numri_blerjeve_cc, dtype: int64

In [47]:
data.groupby('Numri_blerjeve_cc')['Shuma_transaksionit_metemadh'].sum().sort_values(ascending=False)

Numri_blerjeve_cc
1      148874.57
12     129210.17
8      115660.88
2      100457.50
14      99240.64
         ...    
142        53.00
254        50.00
0           0.24
199         0.00
347         0.00
Name: Shuma_transaksionit_metemadh, Length: 173, dtype: float64

In [48]:
data.groupby('CREDIT_LIMIT')['CUST_ID'].sum().sort_values(ascending=False).head()

CREDIT_LIMIT
1833.333333    C18715
2350.000000    C18480
1450.000000    C18417
650.000000     C17714
3777.777778    C17586
Name: CUST_ID, dtype: object

In [51]:
data.groupby('CUST_ID')['Shuma_transaksionit_metemadh'].sum().sort_values(ascending=False)

CUST_ID
C10574    40761.25
C10523    40624.06
C14048    34087.73
C11657    33803.84
C10284    26547.43
            ...   
C14539        0.00
C14537        0.00
C14536        0.00
C14535        0.00
C14599        0.00
Name: Shuma_transaksionit_metemadh, Length: 8950, dtype: float64

Which client did the most/biggest transaction transaction

In [52]:
data

CUST_ID  Bilanci_aktual  Frekuenca_ndryshimit_bilancit  \
0     C10001       40.900749                       0.818182   
1     C10002     3202.467416                       0.909091   
2     C10003     2495.148862                       1.000000   
3     C10004     1666.670542                       0.636364   
4     C10005      817.714335                       1.000000   
...      ...             ...                            ...   
8945  C19186       28.493517                       1.000000   
8946  C19187       19.183215                       1.000000   
8947  C19188       23.398673                       0.833333   
8948  C19189       13.457564                       0.833333   
8949  C19190      372.708075                       0.666667   

      Shuma_blerjeve_nga_llogaria  Shuma_transaksionit_metemadh  \
0                           95.40                          0.00   
1                            0.00                          0.00   
2                          773.17                        773.17   
3                         1499.00                       1499.00   
4                           16.00                         16.00   
...                           ...                           ...   
8945                       291.12                          0.00   
8946                       300.00                          0.00   
8947                       144.40                          0.00   
8948                         0.00                          0.00   
8949                      1093.25                       1093.25   

      Shuma_si_kest  Frekuenca_perdorimit_cc  Numri_blerjeve_cc  CREDIT_LIMIT  \
0             95.40                 0.166667                  2        1000.0   
1              0.00                 0.000000                  0        7000.0   
2              0.00                 1.000000                 12        7500.0   
3              0.00                 0.083333                  1        7500.0   
4              0.00                 0.083333                  1        1200.0   
...             ...                      ...                ...           ...   
8945         291.12                 1.000000                  6        1000.0   
8946         300.00                 1.000000                  6        1000.0   
8947         144.40                 0.833333                  5        1000.0   
8948           0.00                 0.000000                  0         500.0   
8949           0.00                 0.666667                 23        1200.0   

      Pjesa_paguar_limitit  Pagesa_minimale_cdomuaj_cc  Kohezgjatja_cc  
0               201.802084                  139.509787              12  
1              4103.032597                 1072.340217              12  
2               622.066742                  627.284787              12  
3                 0.000000                         NaN              12  
4               678.334763                  244.791237              12  
...                    ...                         ...             ...  
8945            325.594462                   48.886365               6  
8946            275.861322                         NaN               6  
8947             81.270775                   82.418369               6  
8948             52.549959                   55.755628               6  
8949             63.165404                   88.288956               6  

[8950 rows x 12 columns]

In [55]:
data.groupby(['Shuma_blerjeve_nga_llogaria', 'Shuma_si_kest']).size().max()

2042

In [60]:
data.groupby(['CREDIT_LIMIT','Pjesa_paguar_limitit'])['CUST_ID'].sum().reset_index()

CREDIT_LIMIT  Pjesa_paguar_limitit CUST_ID
0             50.0           7570.207782  C17140
1            150.0            119.626820  C12567
2            150.0            298.102350  C16744
3            150.0            316.324154  C16725
4            150.0            699.292482  C15662
...            ...                   ...     ...
8756       23000.0           8760.860264  C15407
8757       25000.0           1389.167981  C13762
8758       28000.0          26557.097240  C11004
8759       30000.0           5772.813158  C15041
8760       30000.0          30029.014960  C17237

[8761 rows x 3 columns]

This shows for each costumer what is the limit of credit and the payed kest of them.